app/Controllers/UpdateController.php

```
@@ -240,23 +242,119 @@ public function downloadBackup(Request $request, Response $response, mysqli $db)
-        if ($content === false) {
+        // Stream the file instead of loading it into memory — backups can be large.
+        $handle = fopen((string) $result['path'], 'rb');
+        if ($handle === false) {
             return $this->jsonResponse($response, ['error' => __('Impossibile leggere il file di backup')], 500);
         }
+        $size = filesize((string) $result['path']);
 
-        $response->getBody()->write($content);
         return $response
-            ->withHeader('Content-Type', 'application/sql')
+            ->withBody(new \Slim\Psr7\Stream($handle))
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '258', column '28'). (undefined)

(MissingImport)